docs/public/components/common-patterns.md

```diff
@@ -0,0 +1,49 @@
+# Common patterns
+
+TODO: A page documenting common pattern:
+
+* Groups
+* Optional
+* Enable/Disable state
+* ...
+
+## Groups
+
+`Fx` [groups](https://pkg.go.dev/go.uber.org/fx#hdr-Value_Groups) are a useful feature that make it easier to produce and consume many values of the same type. A component
+can add any type into groups which can be consumed by other components.
+
+ 
+
+For example:
+
+Here, two components add a `server.Endpoint` type to the `server` group (note the `group` label in the `fx.Out` struct).
+
+=== ":octicons-file-code-16: todolist/todolist.go"
+    ```go
+    type provides struct {
+        fx.Out
```

Comment:
If we're targeting the new layout should this use compdef.Out

Reply:
This page is in TODO, I just imported from the previous doc.